main.go

```diff
@@ -2,40 +2,67 @@ package main
 
 import (
 	"flag"
+	"fmt"
 	"math"
 	"math/rand"
 	"os"
 	"strconv"
+	"sync"
 	"time"
 
-	"github.com/elastic/hey-apm/benchmark"
+	"go.elastic.co/apm"
 
+	"github.com/elastic/hey-apm/benchmark"
 	"github.com/elastic/hey-apm/models"
-
 	"github.com/elastic/hey-apm/worker"
 )
 
+var r *rand.Rand
+
+func init() {
+	apm.DefaultTracer.Close()
+	r = rand.New(rand.NewSource(1000))
```

Comment:
Indeed the fixed seed is for repeatability. When running the workload a couple of times against different endpoints, having a fixed seed is desirable.

Reply:
Fair enough. I'd suggest just using `math/rand.Intn` then, since it uses a locked `math/rand.Source` internally. If you created a separate Rand to avoid other users of the source affecting repeatability, then a mutex should be used.

Reply:
fully agree, thanks for the review; I just haven't pushed the changes yet.